main.go

```diff
@@ -97,28 +98,40 @@ type config struct {
 	GenericWebhookURLs []string `yaml:"generic_webhook_urls"`
 }
 
-// RejectionCondition contains the fields that should match a bug report for it to be rejected.
+// RejectionCondition contains the fields that can match a bug report for it to be rejected.
 type RejectionCondition struct {
-	// Required field: if a payload does not match this app name, the condition does not match.
+	// All the optional fields must match for the rejection condition to apply
+	// App name, checked only if not empty
 	App string `yaml:"app"`
-	// Optional: version that must also match in addition to the app and label. If empty, does not check version.
+	// Version, checked only if not empty
 	Version string `yaml:"version"`
-	// Optional: label that must also match in addition to the app and version. If empty, does not check label.
+	// List of labels, checked only if not empty
 	Label string `yaml:"label"`
+	// Message sent by the user, checked only if not empty
+	UserTextMatch string `yaml:"usertext"`
+	// Send this text to the client-side to inform the user why the server rejects the rageshake. Uses a default generic value if empty.
+	Reason string `yaml:"reason"`
 }
 
-// shouldReject returns true if the app name AND version AND labels all match the rejection condition.
-// If any one of these do not match the condition, it is not rejected.
-func (c RejectionCondition) shouldReject(appName, version string, labels []string) bool {
-	if appName != c.App {
-		return false
+// shouldReject returns true if all the App, Version, Label and UserTextMatch attributes of a RejectionCondition match
+func (c RejectionCondition) shouldReject(appName, version string, labels []string, userText string) (reject bool, reason *string) {
```

Comment:
I think it woldl be cleaner to take a `*payload` rather than have this ever-growing list of parameters.

Reply:
Done, I also moved the version extraction logic into `shoudReject` from `matchesRejectionCondition` to do it cleanly.